heuristic/repair_operators.py

```diff
@@ -324,228 +218,120 @@ def worst_employee_repair(
+            delta_calculate_negative_deviation_from_contracted_hours(state, [e], contracted_hours, weeks, time_periods_in_week, competencies, time_step)
             calculate_weekly_rest(state, shifts_in_week, [e], weeks)
             calculate_partial_weekends(state, [e], shifts_at_day, saturdays)
             calculate_isolated_working_days(state, [e], shifts_at_day, days)
             calculate_isolated_off_days(state, [e], shifts_at_day, days)
             calculate_consecutive_days(state, [e], shifts_at_day, L_C_D, days)
-
-            # Hard restriction:
-            below_minimum_demand(
-                state,
-                repaired,
-                employee_with_competencies,
-                demand,
-                competencies,
-                t_covered_by_shift,
-            )
-            above_maximum_demand(
-                state,
-                repaired,
-                employee_with_competencies,
-                demand,
-                competencies,
-                t_covered_by_shift,
-            )
+            
+            #Hard restriction:
+            below_minimum_demand(state, repaired, employee_with_competencies, demand, competencies, t_covered_by_shift)
+            above_maximum_demand(state, repaired, employee_with_competencies, demand, competencies, t_covered_by_shift)
             more_than_one_shift_per_day(state, [e], demand, shifts_at_day, days)
             cover_multiple_demand_periods(state, repaired, t_covered_by_shift, competencies)
-            mapping_shift_to_demand(
-                state, repaired, t_covered_by_shift, shifts_overlapping_t, competencies
-            )
-
-            # Stores the objective function for this employee
-            employee_objective_functions[e] = regret_objective_function(
-                state,
-                e,
-                off_shifts,
-                saturdays,
-                days,
-                L_C_D,
-                weeks,
-                contracted_hours,
-                competencies,
-                [shift[0]],
-            )
-            # Is needed to set the decision variable back to 0
-            set_x(state, t_covered_by_shift, e, shift[0], shift[1], 0)
-
-        # if(len(employee_objective_functions.keys()) == 0):
-        #   return repair_set
+            mapping_shift_to_demand(state, repaired, t_covered_by_shift, shifts_overlapping_t, competencies)
+            
+            competency_score = score - len(competencies_needed)
+            #Stores the objective function for this employee
+            employee_objective_functions[e] = regret_objective_function(state, e, off_shifts, saturdays, days, L_C_D, weeks, contracted_hours, competencies, [shift[0]], competency_score)
+            #Is needed to set the decision variable back to 0
+            set_x(state, t_covered_by_shift, e, shift[0], shift[1], 0, y_s)
+        
+        #if(len(employee_objective_functions.keys()) == 0):
```

Comment:
Remove

Reply:
This if-statement is what keeps the function from throwing a valueError (think this was the right error). Should be kept and used if we continue to see this problem. I commented it out to see if the problem was solved other places. I have not had any issues with it in my branch, but could be needed later on.